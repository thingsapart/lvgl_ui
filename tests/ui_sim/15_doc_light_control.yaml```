# TICKS: 5
- type: data-binding
  state:
    - light_on: false
    - brightness: 100.0 # As a percentage
  actions:
    - toggle_light:
        toggle: light_on
    - set_brightness:
        set: { brightness: [clamp, value.float, 0, 100] }
  schedule:
    - { tick: 1, action: toggle_light }
    - { tick: 2, action: set_brightness, with: 50 }
    - { tick: 3, action: set_brightness, with: 120 }
    - { tick: 4, action: set_brightness, with: -10 }
    - { tick: 5, action: toggle_light }
