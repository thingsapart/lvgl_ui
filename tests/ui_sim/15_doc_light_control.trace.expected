--- UI-Sim Trace Start ---
STATE_SET: brightness = 100.000 (old: null)
STATE_SET: light_on = false (old: null)
NOTIFY: brightness = 100.000
NOTIFY: light_on = false

--- TICK 1 ---
ACTION: toggle_light value=null
STATE_SET: light_on = true (old: false)
NOTIFY: light_on = true

--- TICK 2 ---
ACTION: set_brightness value=50.000
STATE_SET: brightness = 50.000 (old: 100.000)
NOTIFY: brightness = 50.000

--- TICK 3 ---
ACTION: set_brightness value=120.000
STATE_SET: brightness = 100.000 (old: 50.000)
NOTIFY: brightness = 100.000

--- TICK 4 ---
ACTION: set_brightness value=-10.000
STATE_SET: brightness = 0.000 (old: 100.000)
NOTIFY: brightness = 0.000

--- TICK 5 ---
ACTION: toggle_light value=null
STATE_SET: light_on = false (old: true)
NOTIFY: light_on = false

--- UI-Sim Trace End ---
